src/error.ts

```diff
@@ -1246,11 +1253,16 @@ export function needsRetryableWriteLabel(error: Error, maxWireVersion: number):
   }
 
   if (error instanceof MongoWriteConcernError) {
-    return RETRYABLE_WRITE_ERROR_CODES.has(error.result.writeConcernError.code ?? error?.code ?? 0);
+    if (!(serverType === 'Mongos' && maxWireVersion < 9)) {
+      return RETRYABLE_WRITE_ERROR_CODES.has(
+        error.result.writeConcernError.code ?? Number(error.code) ?? 0
+      );
+    }
+    return RETRYABLE_WRITE_ERROR_CODES.has(Number(error.originalCode) ?? 0);
```

Comment:
```suggestion
    return RETRYABLE_WRITE_ERROR_CODES.has(Number(error.result.code) ?? 0);
```

just a thought since the top-level code is already available on the result document.  

If you'd prefer to leave `originalCode` on the write concern error, can we add the originalCode property to the write concern error class as an internal field?  This works right now because MongoError has `[key: string]: any` in it, so it allows any keys.  But if we expect to use `originalError`, we should explicitly add it to our error type.

Reply:
I see, I'm good with using `error.result.code` here instead of adding a property explicitly.